### ISSUE TITLE
Migration concept / data import from other chats (HipChat, Mattermost, ...)

### ISSUE THREAD

**Issue by nesretep-anp1** [matrix-org/synapse#4590]
**Description:**

I had intense tests with Synapse/Riot the last days where I got all (wanted) things the way I/we would want them! Nice work so far. (Ok, having the status.im theme on iPhone would be great, because the currently built theme are awfull, but that's a different story).

We want to leave Mattermost in favour of Matrix/Synapse. So, we now would have to migrate our Mattermost data (in)to Synapse (more than 250000 messages). But I did not find any migration concept or tool to do that?!

**Comment by t3chguy** [comment]
It is definitely possible, you could use APIs like https://matrix.org/docs/spec/application_service/r0.1.0.html#timestamp-massaging to even maintain the exact timestamps.

**Comment by nesretep-anp1** [comment]
Nice! ;-)

But I do not want to develop a migration/import tool! ;-) The question is: is there an appropriate tool?

**Comment by Half-Shot** [comment]
@nesretep-anp1 Modular.im has a hipchat migration tool built but there isn't a mattermost one yet.

**Comment by neilisfragile** [comment]
@nesretep-anp1 I don't think we we can invest in a mattermost migrator immediately (though obviously it is something we would like to have) If you are considering modular.im that would definitely help us bump it up the to do list.

**Comment by nesretep-anp1** [comment]
> 
> 
> @nesretep-anp1 I don't think we we can invest in a mattermost migrator immediately (though obviously it is something we would like to have) If you are considering modular.im that would definitely help us bump it up the to do list.

Well, to be honest; wie DO use Mattermost (und DO evaluate Synapse) due to having the appropriate communications platform within OUR OWN data center. So using modular.im is not an option in any case. (I think this will be the rule for nearly all leaving HipChat customers as well).

But, if there is a bounty based way to get an functional migration tool for Mattermost, that we would be ready to discuss!? ;)

**Comment by nesretep-anp1** [comment]
> But, if there is a bounty based way to get an functional migration tool for Mattermost, that we would be ready to discuss!? ;)

No one?

**Comment by hamoid** [comment]
I'm in the same situation. Our community will be stuck with Mattermost until there's a way to migrate to Matrix.

I see there's an export tool for Mattermost: https://docs.mattermost.com/administration/bulk-export.html But exporting file attachments is apparently not supported yet. This seems to be the command:

     ./mattermost export bulk /home/user/bulk_data.json --all-teams

**Comment by richvdh** [comment]
related: #3716

**Comment by gabrc52** [comment]
As for Mattermost, I am working on a set of scripts to help with migrating to Matrix.

https://github.com/gabrc52/mattermost-to-matrix-migration

For now I have `download_everything.sh` which will download all public messages in a Mattermost team, including media. I still need to write the other half.
